diagnostic.md

```diff
@@ -5,74 +5,101 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```bash
-// your response here
+The purpose of the back end is to take the requests from the front end, complete
+the processes involved with those requests (i.e. update a table or delete a line
+item), and then to save that change in the database so that change will be
+reflected in the future. After all of this the backend sends the front end
+the requested information. The backend communicates with servers in order to
+track information and change data.
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```bash
-// your response here
+The model layer should fetch the data, the controller layer should query the
+model for that data.
 ```
 
 Which layer in the MVC pattern communicates with the model?
 
 ```bash
-// your response here
+The conroller layer communicates with the model. The controller instructs
+the model to make the approrpiate changes and then passes the info to the
+view layer to be disaplyed after the model layer has completed its processes.
 ```
 
 Why don't we use views in our interpretation of the MVC pattern?
 
 ```bash
-// your response here
+Because views are simiply the visualization of the information, they dont
+contribute much in regards to information processing or conversaion.
+They simiply display the information given to them by the controller.
 ```
 
 What does C.R.U.D stand for?
 
 ```bash
-// your response here
+Create
+Read
+Update
+Delete
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
-```bash
-// your response here
+```bashcd
+Controller
 ```
 
 List at least 5 standard rails actions that C.R.U.D requests correspond to?
 
 ```bash
-// your response here
+GET
+POST
+PATCH
+DELETE
+UPDATE
 ```
 
 A user action fires a `GET` request for `/people/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```bash
-// your response here
+1. The server must be expecting the data in a certain format. If the request is
+sent to the server in the proper format, the server can then begin processing
+the request.
+
+2. Next, the server looks for the appropriate page value ('/people/') and then looks for
+the appropriate value ('1').
+
+3. Should the server find both, it will then construct the URL and send it back
+to the user in the form of a page. 
 ```
 
 What is the command to generate a new rails-api app?
 
 ```bash
-// your response here
+rails g scaffold
 ```
```

Comment:
Almost! scaffold is for building out rails components

this would be just `rails-api new <appname>` or `rails new <appname> --api`